### PR TITLE
Unit Tests For Event Service and DB Mocking

### PR DESCRIPTION
**Pull request by Jamie-Unosquare**
Added Functionality to allow EntityFrameworkContext and Repository Mocking
String interpolation and removal of unused method from Event Service
Added missing info from MockDatabase

### CODE REVIEW

**Comment by Jamie-Unosquare** [comment]
Currently a lot of the expected results are just written into the asserts as they're set up in the MockDatabase class. Would like opinions on if this is fine or if I should add a variable that queries the mockdatabase to get the actual number for example querying all events for annualLeave to compare for the necessary test.

**Comment by BrownMark** [comment]
To be honest Jamie, what you're mentioning is my reservation with the mock database approach. I personally prefer the arrangements for a unit test to happen in the unit test. Makes maintenance much more straightforward and less likely to break other unit tests.

**Comment by Jamie-Unosquare** [comment]
Do you have any suggestions to what would be the best way to do this?

File: AdminCore.Services.Tests/MockDatabase.cs [hunk 1]
```diff
@@ -17,6 +17,7 @@ public class MockDatabase
     public List<EventStatus> EventStatusRepository;
     public List<EventType> EventTypeRepository;
     public List<Team> TeamRepository;
+    public List<MandatoryEvent> MandatoryEventRepository;
```
Comment (Jamie-Unosquare) [inline AdminCore.Services.Tests/MockDatabase.cs:20]: These changes can be ignored for now as this class will be deleted when DashboardService tests are updated

**Review by BrownMark — APPROVED** [review]

**Pull request merged**